cirq/ops/raw_types.py

```diff
@@ -327,6 +327,14 @@ def _decompose_(self, qubits):
         return inverse.inverse(decompose.decompose_once_with_qubits(
             self._original, qubits))
 
+    def _has_unitary_(self):
+        from cirq import line
+        qubits = line.LineQubit.range(self._num_qubits_())
+        return all(
+            protocols.has_unitary(e)
+            for e in decompose.decompose_once_with_qubits(
+                self._original, qubits))
```

Comment:
Is it guaranteed that `cirq.has_unitary(gate) == cirq.has_unitary(cirq.inverse(gate))`?

A more robust version would be to replace `self._original` with `self` here (but less efficient).

Reply:
Acknowledged, but for unitary operations it must be the case that the inverse is unitary if the main is unitary. Technically someone can write a class where that's not the case, but it would be an incorrect class.